File: server/src/main/java/com/mthwate/conk/listener/JumpListener.java
package com.mthwate.conk.listener;

import com.jme3.network.HostedConnection;
import com.mthwate.conk.message.JumpMessage;
import com.mthwate.conk.user.UserStore;

/**
 * @author mthwate
 */
public class JumpListener extends AbstractServerListener<JumpMessage> {

	@Override
	protected void onReceived(HostedConnection source, JumpMessage m) {
		UserStore.getUser(source).getControl().jump();
	}

}

File: server/src/main/kotlin/com/mthwate/conk/listener/JumpListener.kt
package com.mthwate.conk.listener

import com.mthwate.conk.message.JumpMessage
import com.mthwate.conk.user.UserStore

/**
 * @author mthwate
 */
class JumpListener : LazyServerListener<JumpMessage>({ src, msg -> UserStore.getUser(src)?.control?.jump() })